### PR TITLE
Add string-get-token primitive.

### PR DESCRIPTION
**Pull request by kjmcqueen**
https://rchain.atlassian.net/browse/ROS-253

Please have mercy on me.  This primitive is complex and exhibits some weird edge condition behaviors.

### CODE REVIEW

File: roscala/src/test/scala/coop/rchain/rosette/prim/RblStringSpec.scala [hunk 1]
```diff
@@ -745,4 +745,63 @@ class RblStringSpec extends FlatSpec with Matchers {
     stringMemQ.fn(newCtxt) should be('left)
   }
 
+  /** string-get-token */
+  "string-get-tokan" should "return a token with specified delimiters" in {
```
Comment (tymm) [inline roscala/src/test/scala/coop/rchain/rosette/prim/RblStringSpec.scala]: Typo.

File: roscala/src/main/scala/coop/rchain/rosette/prim/rblstring.scala [hunk 2]
```diff
@@ -207,6 +207,53 @@ object rblstring {
...
+    @checkArgumentMismatch
+    override def fn(ctxt: Ctxt): Either[PrimError, RblString] = {
+      val elem = ctxt.argvec.elem
+
+      for {
+        str <- checkType[RblString](0, elem) // Ensure arg0 is a RblString
+        w   <- checkType[Fixnum](1, elem)    // Ensure arg1 is a Fixnum
+        sep <- checkType[RblString](2, elem) // Ensure arg2 is a RblString
+      } yield {
+        // The C++ Rosette exhibits some weird edge conditions.
```
Comment (tymm) [inline roscala/src/main/scala/coop/rchain/rosette/prim/rblstring.scala]: Can you give us some examples?

Comment (kjmcqueen) [reply roscala/src/main/scala/coop/rchain/rosette/prim/rblstring.scala]: I started to get these and realized I didn't understand some aspects of how it operates.  I'm going to cancel the PR and work on it a while longer.  Sorry.

**Comment by kjmcqueen** [comment]
There are some aspects of this primitive I don't understand yet. Closing for now.
